### ISSUE TITLE
Password change in settings very broken

### ISSUE THREAD

**Issue by oscardssmith** [pop-os/cosmic-settings#844]
Changing your password from the system and accounts menu seems deeply broken. Specifically, it appears to not actually change your password (and sometimes it seems like it might change your password to the wrong password).

**Comment by bhh32** [comment]
Can you give more details, specifically abut the changing it to the wrong password? Where there specific special characters or anything else that it may have interpreted incorrectly?

**Comment by oscardssmith** [comment]
I don't think so. it was all printable ASCII. I'm not entirely sure what happened since I changed it, and the time I realized that something was wrong was after a restart at which point neither the old nor new password seemed to work (although it is definitely possible that I mistyped the old or new password repeatedly)

**Comment by novinxy** [comment]
I had similar experience with changing password. I wrote typo when I changed password and had to do recovery.
There are missing bits in UI/UX like hovers, etc. which might create confusion, here is list of potential issues I see:
- no hover on edit icons & hover on textbox. It might create confusion _"Should I click edit icon or just click on textbox ?"_
- no border for textbox. hard to  tell where input should be
- accepting changes work differently for 'Full name'/'Username' vs Password
   - for Full name & Username you can either press return key or click outside textbox to accept changes. Non technical people might not figure it out, in many cases people just prefer to have 'accept/ok' button. It might be nice addition.
   - password only works by pressing return key, clicking outside textbox doesn't change password. It works in different way and there is no information that change was applied or not. Only one method (hidden one) to accept is insufficient. Some people might know what to do
- to fix issue with writing 'typos' there are 2 simple additions that would help [See examples](https://www.google.com/search?client=firefox-b-d&sca_esv=2074af11bd466d63&q=change+password+dialog&udm=2&fbs=ABzOT_CWdhQLP1FcmU5B0fn3xuWpA-dk4wpBWOGsoR7DG5zJBr1qLlHFB6ZBcx-Arq68_wfw0s-Sy4efUF8x4O2idyUNU17MxNnWP-gPg-bH3QglMxJINyM4i32GPOhiyHOinjE1SNJ-lrwwfWI8tJPY6E6U6Q-vb3O-euuSKgF-383g6tbUXVM&sa=X&ved=2ahUKEwjswf_czcaLAxU5CBAIHd3uAkAQtKgLegQIGBAB&biw=1920&bih=962)
   - 'confirm password' textbox in which we make sure that user wrote password correctly
   - button to 'unhide' password. Without it, it's impossible to review password

I hope this list will serve as nice inspiration for fixes :)

**Comment by neopostmodern** [comment]
I just had a very similar experience and would strongly plea for disabling the whole thing until the UX is fixed. This is so dangerous.

**Comment by cappsyco** [comment]
Was going to raise a bug for this but I believe this issue covers it. From what I can see, new user passwords and changed passwords save in /etc/shadow in plain text, as below:

![Image](https://github.com/user-attachments/assets/aa5db0a8-9abc-44cb-951e-89fdbe78349f)

**Comment by oscardssmith** [comment]
thanks!
